modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/user/session/v1/UserSessionMeSuccessTest.java

```diff
@@ -0,0 +1,158 @@
+ */
+
+package org.wso2.identity.integration.test.rest.api.user.session.v1;
+
+import io.restassured.RestAssured;
+import org.apache.axis2.AxisFault;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpStatus;
+import org.junit.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+
+import java.util.List;
+import javax.xml.xpath.XPathExpressionException;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.core.Is.is;
+
+/**
+ * Test REST API for managing logged in user's sessions.
+ */
+public class UserSessionMeSuccessTest extends UserSessionTest {
+
+    @Factory(dataProvider = "restAPIUserConfigProvider")
+    public UserSessionMeSuccessTest(TestUserMode userMode, String username1, String username2) throws Exception {
+
+        super.init(userMode);
+        this.context = isServer;
+        this.tenant = context.getContextTenant().getDomain();
+        this.session_test_user1 = username1;
+        this.session_test_user2 = username2;
+        this.authenticatingUserName = session_test_user1 + "@" + tenant;
+        this.authenticatingCredential = TEST_USER_PASSWORD;
+    }
+
+    @DataProvider(name = "restAPIUserConfigProvider")
+    public static Object[][] restAPIUserConfigProvider() {
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN, "sessionTestUser1", "sessionTestUser2"}
+//                {TestUserMode.TENANT_ADMIN, "sessionTestUser1", "sessionTestUser2"}
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws XPathExpressionException, AxisFault {
+
+        super.testInit(API_VERSION, swaggerDefinition, tenant);
+        initUrls("me");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testConclude() {
+
+        super.conclude();
+    }
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit() {
+
+        RestAssured.basePath = basePath;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void testFinish() {
+
+        RestAssured.basePath = StringUtils.EMPTY;
+    }
+
+    @Test
+    public void testGetUserSessions() {
+
+        getResponseOfGet(this.sessionsEndpointURI)
+                .then()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .log().ifValidationFails()
+                .body("size()", is(2))
+                .body("userId", notNullValue())
+                .body("sessions", notNullValue())
+                .body("sessions.applications", notNullValue())
+                .body("sessions.applications.subject", notNullValue())
+                .body("sessions.applications.appName", notNullValue())
```

Comment:
can't we validate the exact app name?

Reply:
updated

Reply:
`sessions.applications.appName` is an array of elements. Therefore we can not do the exact app name match since the order of elements can be changed